tests/integration/data_processing/test_data_processor.py

```diff
@@ -53,3 +55,125 @@ def test_data_processor_with_anonymized_columns(tmpdir):
     # Assert
     assert reverse_transformed.occupation.isin(data.occupation).sum() == 0
     assert 'occupation' not in transformed.columns
+
+
+def test_data_processor_with_anonymized_columns_and_primary_key(tmpdir):
+    """Test the ``DataProcessor``.
+
+    Test that when we update a ``pii`` column this uses ``AnonymizedFaker`` to create
+    a new set of data for that column, while it drops it on the ``transform`` and this values
+    can be repeated. Meanwhile, when we set a primary key this has to have a unique length
+    of the input data.
+
+    Setup:
+        - Load metadata and data.
+        - Parse the old metadata to a new metadata.
+        - Anonymize the field ``occupation``.
+        - Create ``id`` and ``ssn`` columns.
+
+    Run:
+        - Create a ``DataProcessor`` with the new metadata.
+        - Fit the ``DataProcessor`` instance.
+        - Transform the data.
+        - Reverse transform the data.
+
+    Side effects:
```

Comment:
I thought we were going to alabel this section `Assert` to match the sections in the test